src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -81,111 +104,187 @@ internal async Task SearchAsync()
+            return true;
+        }
+
+        /// <summary>
+        /// Returns a sequence of groups of projects to process.  The sequence is in priority order,
+        /// and all projects in a particular group should be processed before the next group.  This
+        /// allows us to associate CPU resources in likely areas the user wants, while also still
+        /// allowing for good parallelization.
+        /// </summary>
+        private ImmutableArray<ImmutableArray<Project>> GetOrderedProjectsToProcess()
+        {
+            // If we're only searching the current doc, we don't need to examine anything else but that.
+            if (_searchCurrentDocument)
+            {
+                var project = _currentDocument?.Project;
+                return project == null
+                    ? ImmutableArray<ImmutableArray<Project>>.Empty
+                    : ImmutableArray.Create(ImmutableArray.Create(project));
+            }
+
+            using var result = TemporaryArray<ImmutableArray<Project>>.Empty;
+
+            using var _ = PooledHashSet<Project>.GetInstance(out var processedProjects);
+
+            // First, if there's an active document, search that project first, prioritizing that
+            // active document and all visible documents from it.
+            if (_activeDocument != null)
+            {
+                processedProjects.Add(_activeDocument.Project);
+                result.Add(ImmutableArray.Create(_activeDocument.Project));
+            }
+
+            // Next process all visible docs that were not from the active project.
+            using var buffer = TemporaryArray<Project>.Empty;
+            foreach (var doc in _visibleDocuments)
+            {
+                if (processedProjects.Add(doc.Project))
+                    buffer.Add(doc.Project);
             }
 
-            // Now, process all visible docs that were not from the active project.
-            var tasks = new List<Task>();
-            foreach (var (currentProject, priorityDocs) in visibleDocs.GroupBy(d => d.Project))
+            if (buffer.Count > 0)
+                result.Add(buffer.ToImmutableAndClear());
+
+            // Finally, process the remainder of projects
+            foreach (var project in _solution.Projects)
             {
-                // make sure we only process this project if we didn't already process it above.
-                if (processedProjects.Add(currentProject))
-                    tasks.Add(Task.Run(() => SearchAsync(currentProject, priorityDocs.Where(d => d.Project == currentProject).ToImmutableArray(), seenItems, isFullyLoaded), _cancellationToken));
+                if (processedProjects.Add(project))
+                    buffer.Add(project);
             }
 
-            await Task.WhenAll(tasks).ConfigureAwait(false);
+            if (buffer.Count > 0)
+                result.Add(buffer.ToImmutableAndClear());
+
+            return result.ToImmutableAndClear();
```

Comment:
grouping of projects is desirable because we want to prioritize searching the projects (and esp. the doucments) teh user is currently active in. That's because some people use navto to quickly get to items in tehir own project.  We don't want to spawn a search that is checking every project in parallel, causing local hits to take much longer to find.

Note: this behavior is what we've had for a while.  it does not change.  I just cleaned up the logic here.